pycontrails/models/cocip/cocip.py

```diff
@@ -1075,6 +1081,89 @@ def _simulate_contrail_evolution(self) -> None:
 
             self.contrail_list.append(final_contrail)
 
+    def _maybe_downselect_met_rad(
+        self,
+        met: MetDataset | None,
+        rad: MetDataset | None,
+        latest_contrail: GeoVectorDataset,
+        time_end: np.datetime64,
+    ) -> tuple[MetDataset, MetDataset]:
+        """Downselect ``self.met`` and ``self.rad`` if necessary to cover ``time_end``.
+
+        If current ``met`` and ``rad`` slices to not include ``time_end``, new slices are selected
+        from ``self.met`` and ``self.rad``. Downselection in space will cover
+        - locations of current contrails (``latest_contrail``),
+        - locations of additional contrails that will be loaded from ``self._downwash_flight``
+          before the new slices expire,
+        plus a user-defined buffer.
+        """
+        if met is None or time_end > met.indexes["time"].to_numpy()[-1]:
+            # compute lookahead for future contrails from downwash_flight
+            met_time = self.met.indexes["time"].to_numpy()
+            mask = met_time >= time_end
+            lookahead = np.min(met_time[mask]) if np.any(mask) else time_end
+
+            # create vector for downselection based on current + future contrails
+            future_contrails = self._downwash_flight.filter(
+                (self._downwash_flight["time"] >= time_end)
+                & (self._downwash_flight["time"] <= lookahead),
+                copy=False,
+            )
+            vector = GeoVectorDataset(
+                **{
```

Comment:
The `**` may not be necessary here? (But you might have a reason for wanting it anyway.)

Reply:
https://github.com/contrailcirrus/pycontrails/pull/225/commits/8030b2f544d70774028150b31bbd5c19bad0d6a7